src/io/hdf5_idas.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime
+from pyrocko import trace
+
+
+META_KEYS = {
+    'measure_length': 'MeasureLength',
+    'start_position': 'StartPosition',
+    'spatial_resolution': 'SpatialResolution',
+    'fibre_index': 'FibreIndex',
+    'fibre_length_multiplier': 'FibreLengthMultiplier',
+     #'unit_calibration': 'Unit Calibration (nm)',
+    'start_distance': 'StartDistance',
+    'stop_distance': 'StopDistance',
+    'normalization': 'Normalization',
+    'decimation_filter': 'DecimationFilter',
+    'gauge_length': 'GaugeLength',
+    'norm_offset': 'NormOffset',
+    'source_mode': 'SourceMode',
+    'time_decimation': 'TimeDecimation',
+    'zero_offset': 'ZeroOffset',
+    'p_parameter': 'P',
+    'p_coefficients': 'P_Coefficients',
+    #'idas_version': 'iDASVersion',
+    'precice_sampling_freq': 'PreciseSamplingFrequency',
+    'receiver_gain': 'ReceiverGain',
+    #'continuous_mode': 'Continuous Mode',
+    'geo_lat': 'Latitude',
+    'geo_lon': 'Longitude',
+    'geo_elevation': 'Altitude',
+
+    'channel': None,
+    'unit': 'RawDataUnit' 
+}
+
+
+def get_meta(h5file):
+    """
+    Get metadata from HDF5 file using (almost) the same fields as for the TDMS files.
+    
+    Parameters
+    ----------
+    h5file : HDF5 file object
+        The file to extract metadata from
+    Returns
+    -------
+        Dictionary containing the metadata 
+
+    """
+
+    key_list = list(META_KEYS.keys())
+    val_list = list(META_KEYS.values())
+
+    meta = dict()
+    # the following is based on the PRODML format
+    for field in ['Acquisition',
+                  'Acquisition/Custom/AdvancedUserSettings',
+                  'Acquisition/Custom/SystemSettings',
+                  'Acquisition/Custom/SystemInformation/GPS',
+                  'Acquisition/Custom/UserSettings',
+                  'Acquisition/Raw[0]']:
+        try:
+            field_keys = h5file[field].attrs.keys()
+            for val in val_list:
+                if val in field_keys:
+                    meta[key_list[val_list.index(val)]] = f[field].attrs[val]
+        except:
```

Comment:
Please use an explicit exception. `KeyError`?

Reply:
I removed this try-except statement, which should work for files with complete header information.